modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -101,6 +130,17 @@ public boolean isSlotActive() {
         return isSlotActive;
     }
 
+    public boolean isAnOverlappingSlot() {
+        return isAnOverlappingSlot;
+    }
+
+    public void setAnOverlappingSlot(boolean isAnOverlappingSlot) {
+        this.isAnOverlappingSlot = isAnOverlappingSlot;
+        if(isAnOverlappingSlot) {
+            addState(SlotState.OVERLAPPED);
```

Comment:
Would it be meaningful to add debug logs here ? Mean to have a way to track each of the states represented by the slot